addons/mail/static/src/models/message/message.js

```diff
@@ -111,8 +111,14 @@ function factory(dependencies) {
                     data2.parentMessage = insertAndReplace(this.convertData(data.parentMessage));
                 }
             }
-            if ('partner_ids' in data && this.messaging.currentPartner) {
-                data2.isCurrentPartnerMentioned = data.partner_ids.includes(this.messaging.currentPartner.id);
+            if ('recipients' in  data) {
+                if (!data.recipients) {
+                    data2.recipients = unlinkAll();
+                } else {
+                    data2.recipients = insertAndReplace(data.recipients.map(recipient =>
+                        this.messaging.models['mail.partner'].convertData(recipient)
+                    ));
+                }
```

Comment:
This if-else is already built in `insertAndReplace` (empty array should be supported).
Also note that in JS to check empty array you need to check `.length`, the condition as it is now is useless.